frontend/app/micro-app.tsx

```diff
@@ -374,13 +375,26 @@ const MicroApp = () => {
     }
   };
 
-  // Function to open URL using browser
-  const handleOpenUrlFromBrowser = async (url: string) => {
+  // Function to open a URL using the browser
+  const handleOpenUrlInBrowser = async (config: BrowserConfig) => {
     try {
-      const result = await WebBrowser.openBrowserAsync(url, {
-        presentationStyle: WebBrowser.WebBrowserPresentationStyle.FULL_SCREEN,
-        enableBarCollapsing: false,
-        dismissButtonStyle: "close",
+      if (!config) {
+        console.error("Missing configs.");
```

Comment:
The error seems a bit too generic
```suggestion
        console.error("Missing Required WebBrowser configuration.");
```